account_operating_unit/models/account_invoice.py

```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
                                         states={'draft': [('readonly',
                                                            False)]})
 
+    @api.onchange('operating_unit_id')
+    def _onchange_operating_unit(self):
+        if self.operating_unit_id:
+            res = self.env['account.journal'].search(
+                [('type', '=', self.journal_id.type)])
+            rf = res.filtered(
+                    lambda aj: aj.operating_unit_id == self.operating_unit_id)
```

Comment:
Please change to `
journal = self.env['account.journal'].search([('type', '=', self.journal_id.type), ('operating_unit_id', '=', self.operating_unit_id.id), limit=1]
self.journal_id = journal
`

Reply:
But if there is no specific journal for the OU, we can take a journal that is not assigned to any OU.

Reply:
As @aheficent said, that way we could not take a journal without an assigned OU.